### PR TITLE
chore(gatsby): Migrate types/node-interface.js to TypeScript

### PR DESCRIPTION
**Pull request by alisson-suzigan**
### Description
This PR converts the file packages/gatsby/src/schema/types/node-interface.js to TS.

### Related Issue
Related to #21995

### CODE REVIEW

**Comment by alisson-suzigan** [comment]
Hi folks.
Does this PR still make sense?

**Comment by alisson-suzigan** [comment]
Hello, is anyone there?

File: packages/gatsby/src/schema/types/node-interface.ts [hunk 1]
```diff
@@ -28,7 +37,7 @@ const getOrCreateNodeInterface = schemaComposer => {
       id: `ID!`,
       parent: {
         type: `Node`,
-        resolve: (source, args, context, info) => {
+        resolve: (source, _args, context): Node => {
```
Comment (vladar) [inline packages/gatsby/src/schema/types/node-interface.ts:40]:
```suggestion
        resolve: (source, _args, context): Node | null => {
```

File: packages/gatsby/src/schema/types/node-interface.ts [hunk 2]
```diff
@@ -40,7 +49,7 @@ const getOrCreateNodeInterface = schemaComposer => {
       },
       children: {
         type: `[Node!]!`,
-        resolve: (source, args, context, info) => {
+        resolve: (source, _args, context): Node => {
```
Comment (vladar) [inline packages/gatsby/src/schema/types/node-interface.ts:52]:
```suggestion
        resolve: (source, _args, context): Array<Node> => {
```

**Review by vladar — CHANGES_REQUESTED** [review]
Thanks for suggesting this 👍 

In general it looks good to me but there are two incorrect types which I mention in inline comments. Fix those and we are good to go!

**Comment by alisson-suzigan** [comment]
@vladar, the code was changed following your suggestion. Thanks.

**Review by vladar — APPROVED** [review]
Awesome, thanks! 💜

**Pull request merged**